metadata-ingestion/tests/unit/test_transform_dataset.py

```diff
@@ -2841,3 +2844,188 @@ def test_add_dataset_data_product_transformation():
+        "email_id@example.co.in",
+        "email_id@example.co.uk",
+        "email_test:XYZ@example.com",
+        "email_id:id1@example.com",
+        "email_id:id2@example.com",
+    ]
+    expected_owner_urns: List[str] = []
+    for user in expected_user_emails:
+        expected_owner_urns.append(
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+    _test_clean_owner_urns(pipeline_context, in_owner_urns, config, expected_owner_urns)
+
+    # remove multiple values
+    config = [":ABCDEF", "email"]
+    expected_user_emails = [
+        "_id@example.com",
+        "123_id@example.com",
+        "_id@example.co.in",
+        "_id@example.co.uk",
+        "_test:XYZ@example.com",
+        "_id:id1@example.com",
+        "_id:id2@example.com",
+    ]
+    expected_owner_urns = []
+    for user in expected_user_emails:
+        expected_owner_urns.append(
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+    _test_clean_owner_urns(pipeline_context, in_owner_urns, config, expected_owner_urns)
+
+    # remove values using RegEx
+    config = [r"(?<=_)(\w+)"]
+    expected_user_emails = [
+        "ABCDEF:email_@example.com",
+        "ABCDEF:123email_@example.com",
+        "email_@example.co.in",
+        "email_@example.co.uk",
+        "email_:XYZ@example.com",
+        "email_:id1@example.com",
+        "email_:id2@example.com",
+    ]
+    expected_owner_urns = []
+    for user in expected_user_emails:
+        expected_owner_urns.append(
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+    _test_clean_owner_urns(pipeline_context, in_owner_urns, config, expected_owner_urns)
+
+    # remove number from URN
+    config = [r"\d+"]
+    expected_user_emails = [
+        "ABCDEF:email_id@example.com",
+        "ABCDEF:email_id@example.com",
+        "email_id@example.co.in",
+        "email_id@example.co.uk",
+        "email_test:XYZ@example.com",
+        "email_id:id@example.com",
+        "email_id:id@example.com",
+    ]
+    expected_owner_urns = []
+    for user in expected_user_emails:
+        expected_owner_urns.append(
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+    _test_clean_owner_urns(pipeline_context, in_owner_urns, config, expected_owner_urns)
+
+    # remove 'example.*
+    config = [r"@example\.\S*"]
+    expected_user_emails = [
+        "ABCDEF:email_id",
+        "ABCDEF:123email_id",
+        "email_id",
+        "email_id",
+        "email_test:XYZ",
+        "email_id:id1",
+        "email_id:id2",
+    ]
+    expected_owner_urns = []
+    for user in expected_user_emails:
+        expected_owner_urns.append(
+            builder.make_owner_urn(user, owner_type=builder.OwnerType.USER)
+        )
+    _test_clean_owner_urns(pipeline_context, in_owner_urns, config, expected_owner_urns)
+
+    # remove XYZ
+    config = ["(?<=:)[A-Z]+(?=@)"]
```

Comment:
Why does the comment say `remove XYZ`? Is the regex only for XYZ or capital alphanumeric in certain cases?

Reply:
updated the comment, here we are removing any CAPITAL letters between `:` and `@`